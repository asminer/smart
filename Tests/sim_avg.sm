


#SeedRng 123456789

print("\n----------------------------------------------------------------\n");
rand int zero := 0;
print("Average for const(0): ", avg(zero), "\n");

print("\n----------------------------------------------------------------\n");
rand int one := 1;
print("Average for const(1): ", avg(one), "\n");

print("\n----------------------------------------------------------------\n");
rand int inf := infinity;
print("Average for const(oo): ", avg(inf), "\n");

print("\n----------------------------------------------------------------\n");
rand int die := equilikely(1,6);
print("Average for equilikely(1,6): ", avg(die), "\n");

print("\n----------------------------------------------------------------\n");
rand int dice := equilikely(1,6) + equilikely(1,6);
print("Average for dice: ", avg(dice), "\n");

print("\n----------------------------------------------------------------\n");
rand int coin := bernoulli(0.6);
print("Average for bernoulli(0.6): ", avg(coin), "\n");

print("\n----------------------------------------------------------------\n");
rand int howmany := geometric(0.6);
print("Average for geometric(0.6): ", avg(howmany), "\n");

print("\n----------------------------------------------------------------\n");
rand int pascal := geometric(0.6) + geometric(0.6) + geometric(0.6);
print("Average for pascal(3, 0.6): ", avg(pascal), "\n");

print("\n----------------------------------------------------------------\n");
rand int coins := binomial(3, 0.6);
print("Average for binomial(3, 0.6): ", avg(coins), "\n");

print("\n----------------------------------------------------------------\n");
rand int altcoins := bernoulli(0.6) + bernoulli(0.6) + bernoulli(0.6);
print("Average for sum of 3 bernoulli's: ", avg(altcoins), "\n");

print("\n----------------------------------------------------------------\n");
rand real x := expo(5);
print("Average for expo(5): ", avg(x), "\n");

print("\n----------------------------------------------------------------\n");
ph real xx := expo(5);
rand real y := xx;
print("Average for ph real expo(5): ", avg(y), "\n");

print("\n----------------------------------------------------------------\n");
rand real earl := erlang(3, 5);
print("Average for erlang(3, 5): ", avg(earl), "\n");


print("\n");
