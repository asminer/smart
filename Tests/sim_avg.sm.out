
----------------------------------------------------------------
Average for const(0): 0 +- 0 (95%)

----------------------------------------------------------------
Average for const(1): 1 +- 0 (95%)

----------------------------------------------------------------
Average for const(oo): null

----------------------------------------------------------------
Average for equilikely(1,6): 3.50286 +- 0.004 (95%)

----------------------------------------------------------------
Average for dice: 6.99739 +- 0.007 (95%)

----------------------------------------------------------------
Average for bernoulli(0.6): 0.600047 +- 0.0006 (95%)

----------------------------------------------------------------
Average for geometric(0.6): 1.50113 +- 0.002 (95%)

----------------------------------------------------------------
Average for pascal(3, 0.6): 4.50204 +- 0.005 (95%)

----------------------------------------------------------------
Average for binomial(3, 0.6): 1.79982 +- 0.002 (95%)

----------------------------------------------------------------
Average for sum of 3 bernoulli's: 1.79981 +- 0.002 (95%)

----------------------------------------------------------------
Average for expo(5): 0.200035 +- 0.0002 (95%)

----------------------------------------------------------------
Average for ph real expo(5): 0.200173 +- 0.0002 (95%)

----------------------------------------------------------------
Average for erlang(3, 5): 0.600019 +- 0.0006 (95%)

